engine/src/externalv0.cpp

```diff
@@ -1441,29 +1454,38 @@ static char *get_array_utf8(const char *arg1, const char *arg2,
                        const char *arg3, int *retval, bool p_is_text)
 {
 	MCarray *value = (MCarray *)arg3;
-	MCVariable *var = NULL;
 	if (MCECptr == NULL)
 	{
 		*retval = xresFail;
 		return NULL;
-	}
-	*retval = trans_stat(getvarptr_utf8(*MCECptr, arg1,&var));
-	if (var == NULL)
-		return NULL;
+    }
     
-	if (!var -> isarray())
-		return NULL;
+    MCContainer var;
+    *retval = trans_stat(getvarptr_utf8(*MCECptr, arg1, var));
+    if (*retval != xresSucc)
+        return NULL;
     
-    MCArrayRef t_array;
-    t_array = (MCArrayRef)var -> getvalueref();
-	if (t_array == nil)
-		return NULL;
+    MCAutoValueRef t_value;
+    if (!var.eval(*MCECptr,
+                  &t_value))
+    {
+        *retval = xresFail;
+        return NULL;
+    }
     
-	if (value->nelements == 0)
-	{
-		value->nelements = MCArrayGetCount(t_array);
-		return NULL;
-	}
+    if (!MCValueIsArray(*t_value))
+    {
+        value->nelements = 0;
+        return NULL;
+    }
+    
+    MCArrayRef t_array = (MCArrayRef)*t_value;
```

Comment:
Well, it is a bad thing as the operation is atomic so the reference count bump is entirely unnecessary. I'd rather we 'fixed' the auto classes to allow movement of a MCValueRef into another auto container or a specific MC<type>Ref container - I'm just not sure what the 'correct' pattern for that is.